cupy/core/fusion.py

```diff
@@ -340,240 +538,186 @@ def res(*args, **kwargs):
             ty_ins = [numpy.dtype(_) for _ in ty_ins]
             ty_outs = [numpy.dtype(_) for _ in ty_outs]
```

Comment:
How about `ty_ins/outs` -> `dtypes_in` and `dtypes_out` respectively (after converted to dtypes)?